### DISCUSSION TITLE
Unable to install on PHP ^7.4 |8.0 Laravel version 8

### DISCUSSION THREAD

**Discussion by urwatech** [spatie/image#144] [Q&A] [answered]
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - league/glide 2.0.x-dev requires php ^5.4 | ^7.0 -> your php version (8.0.10) does not satisfy that requirement.
    - league/glide 2.0.0 requires league/flysystem ^2.0 -> found league/flysystem[2.0.0-alpha.1, ..., 2.x-dev] but the package is fixed to 1.1.5 (lock file version) by a partial update and that version does not match. Make sure you list it as an argument for the update command.
    - spatie/image 2.1.0 requires league/glide ^2.0 -> satisfiable by league/glide[2.0.0, 2.0.x-dev].
    - Root composer.json requires spatie/image ^2.1 -> satisfiable by spatie/image[2.1.0].

Use the option --with-all-dependencies (-W) to allow upgrades, downgrades and removals for packages currently locked to specific versions.

Installation failed, reverting ./composer.json and ./composer.lock to their original content.

**Comment by damms005** [comment, marked as answer]
I have not dug deeper into this, but the reason why I am having OP's error on my machine is because the latest version of Laravel [locked league/glide to `^1.1`](https://github.com/laravel/framework/blob/5398fbbbf81ed842cf0430b278f07eb0f869cf8f/composer.json#L27), while `spatie/image` requires [version `^2.0`](https://github.com/spatie/image/blob/bf5afcac04b24d9973e62e7ca63519e2f21a2d62/composer.json#L23) of same.

**Reply by urwatech** [reply]
Yes exactly, but do you know how to fix it?

**Reply by damms005** [reply]
The obvious immediate fix will be to downgrade `spatie/image` to the latest version that depends on v1 of `league/glide`, which will be [v1.10.5](https://github.com/spatie/image/blob/63a963d0200fb26f2564bf7201fc7272d9b22933/composer.json#L23).

Long term fix will be to wait on a version of Laravel that depends on v2 of `league/glide`, or more plausibly, for `spatie/image` to support both versions of glide, like so: `league/glide": "^1.0|^2.0"`

**Reply by urwatech** [reply]
Thanks @damms005 for your support I really appreciate <3

**Reply by seabasss** [reply]
+1 would love support for both versions :)
